src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Threading;
+using Newtonsoft.Json;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Packaging.Licenses;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+
+namespace NuGet.CommandLine
+{
+
+    [Command(typeof(NuGetCommand), "search", "SearchCommandDescription",
+            UsageSummaryResourceName = "SearchCommandUsageSummary", UsageExampleResourceName = "SearchCommandUsageExamples")]
+    public class SearchCommand : Command
+    {
+        private readonly List<string> _sources = new List<string>();
+
+        [Option(typeof(NuGetCommand), "SearchCommandSourceDescription")]
+        public ICollection<string> Source
+        {
+            get { return _sources; }
+        }
+
+        // -Source "https://api.nuget.org/v3/index.json"
+
+
+        [Option(typeof(NuGetCommand), "SearchCommandPreRelease")]
+        public bool PreRelease { get; set; } = false;
+
+
+        [Option(typeof(NuGetCommand), "SearchCommandTake")]
+        public int Take { get; set; } = 20;
+
+
+        private IList<PackageSource> GetEndpointsAsync()
+        {
+            List<PackageSource> configurationSources = SourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
+
+            IList<PackageSource> packageSources;
+            if (Source.Count > 0)
+            {
+                packageSources = Source
+                    .Select(s => PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
```

Comment:
I'd parallelize these.
It'd take forever if you run it on something like NuGet.Client's nuget.config. 

@martinrrm wrote a parallelization method in here: https://github.com/NuGet/NuGet.Client/pull/3480/files#diff-3e1810d5a46cafc4994428412a810de5R20. 

Potentially this can be written in a generic way. 

I think you can do this in a future PR, after Martin's code is merged.

Reply:
@advay26 

Can you please create a follow-up issue about this?

Reply:
Done. https://github.com/NuGet/Home/issues/9772